adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -345,13 +347,31 @@ func (swagger *MgwSwagger) setXWso2Cors() {
 				logger.LoggerOasparser.Errorf("Error while parsing %v: "+err.Error(), xWso2Cors)
 				return
 			}
-			logger.LoggerOasparser.Debugf("Cors Configuration is applied : %+v\n", corsConfig)
-			swagger.xWso2Cors = corsConfig
+			if corsConfig.Enabled {
+				logger.LoggerOasparser.Debugf("API Level Cors Configuration is applied : %+v\n", corsConfig)
+				swagger.xWso2Cors = corsConfig
+				return
+			}
+			swagger.xWso2Cors = generateGlobalCors()
 			return
 		}
 		logger.LoggerOasparser.Errorf("Error while parsing %v .", xWso2Cors)
+	} else {
+		swagger.xWso2Cors = generateGlobalCors()
 	}
+}
 
+func generateGlobalCors() *CorsConfig {
+	conf, _ := config.ReadConfigs()
```

Comment:
shouldn't we handle the error here instead of ignoring it?

Reply:
actually at this point, the config is already read (Coz this is a once operation). We should refactor the readConfigs methods I suppose.